### DISCUSSION TITLE
SQLModel does not support deferred column loading

### DISCUSSION THREAD

**Discussion by MarkusDressel** [fastapi/sqlmodel#1212] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from uuid import UUID
from sqlmodel import Column, Field, LargeBinary
from sqlalchemy.orm import deferred
from sqlmodel import SQLModel, select
from sqlmodel.ext.asyncio.session import AsyncSession
from sqlalchemy.ext.asyncio import AsyncSession, async_sessionmaker, create_async_engine
class Document(SQLModel, table=True):
    id: UUID = Field(
        
        primary_key=True,
        index=True,
        nullable=False,
    )
    name: str = Field(nullable=False)  
    blob: bytes | None = Field(default=None, sa_column=deferred(Column(LargeBinary),raiseload=True))


async def get_document(*,document_id: UUID, db_session: AsyncSession)->Document | None:
    query = select(Document).where(Document.id == document_id)
    response = await db_session.execute(query)
    return response.scalar_one_or_none()

async def create_doc(*,doc:Document, db_session: AsyncSession)->Document:
    
    db_session.add(doc)
    await db_session.commit()
    await db_session.refresh(doc)
    return doc


HOST = 'postgresql+asyncpg://local_user:12345678@localhost:5432/postgres'
engine= create_async_engine(
            url=HOST,
            echo=True,
            future=True,
            # pool_size=POOL_SIZE,
            max_overflow=64,
        )

_sessionmaker = async_sessionmaker(
            autocommit=False,
            autoflush=False,
            bind=engine,
            class_=AsyncSession,
            expire_on_commit=False,
)

async def main():
    byte_obk = b"Hello World"
    doc = Document(id=UUID("c3f2d9d6-1b7c-4c4e-8c1d-1e7b1b9b8e8d"), name="Test", blob=byte_obk)
    async with engine.begin() as conn:
        await conn.run_sync(SQLModel.metadata.drop_all)
        await conn.run_sync(SQLModel.metadata.create_all)
    async with _sessionmaker() as db:
        created_doc = await create_doc(doc=doc, db_session=db)

        # we can see in the echo_engine logs, that a select statement INCLUDING the blob attributes is performed
        document = await get_document(document_id=UUID("c3f2d9d6-1b7c-4c4e-8c1d-1e7b1b9b8e8d"), db_session=db)

        # that should not work, since blob is deferred and is async cannot be loaded
        print(document.blob)
    async with engine.begin() as conn:
        await conn.run_sync(SQLModel.metadata.drop_all)
if __name__ == "__main__":
    import asyncio
    asyncio.run(main())
```

### Description

I would like to declare certain columns of a table as deferred like SQLAlchemy is supporting it like [here](https://docs.sqlalchemy.org/en/14/orm/loading_columns.html#deferred-column-loading) and the when I run select queries like 
```python
select(Model)
```
the deferred blob as not loaded by default unless I add the option to undefer

Setup:
Using SQLModel with async and as postgreSQL database

* Create a table called Document, where one column "blob" should apply deferred loading
* Add a new document to the database
* querying the database to get a the document
* checking that the document has the attribute "blob" not loaded like sqlmodel does for relationships

But in fact, the blob is loaded

Expected result: Blob is not loaded

### Operating System

Linux

### SQLModel Version

0.0.22

### Python Version

Python 3.11.7

**Comment by berryboylb** [comment]
any hope of this getting resolved

**Comment by YuriiMotov** [comment]
There is an issue: https://github.com/fastapi/sqlmodel/issues/97
Let's track it there

The workaround is to defer on query time: https://github.com/fastapi/sqlmodel/issues/97#issuecomment-3184250717
